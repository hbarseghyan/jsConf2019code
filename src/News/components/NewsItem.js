import React from "react";
import PropTypes from "prop-types";

import { Card } from "react-bootstrap";

const NewsItem = ({ name, text }) => {
  return (
    <Card className="customer-card" style={{ width: "16rem" }}>
      <Card.Header>{name}</Card.Header>
      <Card.Body>
        <Card.Text>
          <small className="s-muted">{text}</small>
        </Card.Text>
      </Card.Body>
    </Card>
  );
};

NewsItem.propTypes = {
  name: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export default NewsItem;
