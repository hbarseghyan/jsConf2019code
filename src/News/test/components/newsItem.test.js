import React from "react";
import { shallow } from "enzyme";
import NewsItem from "../../components/NewsItem";

describe("NewsItem", () => {
  const props = { name: "test 1", text: "text 2" };
  it("should render a card", () => {
    expect(shallow(<NewsItem {...props} />)).toMatchSnapshot();
  });
});
