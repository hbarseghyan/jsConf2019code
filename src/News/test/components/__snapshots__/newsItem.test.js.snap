// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsItem should render a card 1`] = `
<Card
  body={false}
  className="customer-card"
  style={
    Object {
      "width": "16rem",
    }
  }
>
  <CardHeader>
    test 1
  </CardHeader>
  <CardBody>
    <CardText>
      <small
        className="s-muted"
      >
        text 2
      </small>
    </CardText>
  </CardBody>
</Card>
`;
